File: frontend/src/services/spendingService.ts
import { api } from "./api";

export const getAllSpending = async () => {
    const response = await api.get("/spending/all");
    return response.data;
};
export const createSpending = async (data : any) => {
    const response = await api.post("/spending/create", data);
    return response.data;
};
export const updateSpending = async (spendingId: number, data : any) => {
    const response = await api.post(`/spending/update/${spendingId}`, data);
    return response.data;
};
export const getSpending = async (spendingId: number) => {
    const response = await api.get(`/spending/${spendingId}`);
    return response.data;
};
export const deleteSpending = async (spendingId: number) => {
    await api.post(`/spending/delete/${spendingId}`);
};
export const getSpendingsByCategory = async(categoryId: number) => {
    const response = await api.get(`spending/category/${categoryId}`);
    return response.data;
}

File: financeTracker/src/main/java/com/springapp/financetracker/service/spending/SpendingService.java
package com.springapp.financetracker.service.spending;

import com.springapp.financetracker.controller.payload.NewSpendingPayload;
import com.springapp.financetracker.dto.SpendingResponseDto;
import com.springapp.financetracker.entity.Spending;
import org.springframework.http.ResponseEntity;

import java.math.BigDecimal;
import java.util.List;
import java.util.Optional;

public interface SpendingService {
    List<SpendingResponseDto> getAllSpending();
    ResponseEntity<SpendingResponseDto> createSpending(NewSpendingPayload payload);
    void updateSpending(Integer id, String name, BigDecimal amount, Integer categoryId);
    ResponseEntity<SpendingResponseDto> findSpending(int spendingId);
    void deleteSpending(int spendingId);
    List<SpendingResponseDto> getSpendingsByCategory(Integer categoryId);
}
